src/main/java/com/nutomic/syncthingandroid/util/AsyncState.java

```diff
@@ -0,0 +1,30 @@
+package com.nutomic.syncthingandroid.util;
+
+import android.os.AsyncTask;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * Tracks the current state of a running AsyncTask without keeping a reference to it.
+ * This allows the object to be GCed as soon as it is finished. Releasing the resources
+ * allocated by the object.
+ */
+public class AsyncState {

```

Comment:
> The instance can not be collected before doInBackground returns because the this reference on the threads stack will keep it alive.
I don't quite understand this, do you mean a reference inside the `AsyncTask` class? Anyway, I think it would be better to make `getEvents()` a private method in `EventProcessor`, so that kind of abstraction won't be needed.